### PR TITLE
feat: Update CustomProperties component to use JSON tokens

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHAT is this pull request doing?

This PR updates the `CustomProperty` component to use the recently added [JSON tokens](https://github.com/Shopify/polaris-react/pull/4809).
- Deleted TypeScript design token objects
- Replaced `tokens` directory with a single `tokens.ts`
  (This simplifies the implementation while still maintaining an opinion on the structure of the tokens object and how it's consumed in TypeScript files)
- Updated `tokens.ts` to import the JSON token files

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

**Comment by BPScott** [comment]
https://github.com/Shopify/polaris-react/blob/main/loom.config.ts#L154 is a demo of adding additional rollup plugins. You'll need to add `@rollup/plugin-json`.

**Review by alex-page — APPROVED** [review]

File: src/tokens/utilities.ts [hunk 1]
```diff
@@ -1,5 +1,19 @@
+import type {TokenGroup} from './tokens';
+
 const BASE_FONT_SIZE = 16;
 
-export function rem(px: string) {
-  return `${parseInt(px, 10) / BASE_FONT_SIZE}rem`;
+function rem(value: string) {
+  return value.replace(
+    // https://regex101.com/r/RBL7EE/1
```
Comment (kyledurand) [inline src/tokens/utilities.ts:7]: These links are temporary right? Not sure it's worth leaving as a comment. Easy enough just to copy and paste and add a few examples

**Review by kyledurand — APPROVED** [review]
🤩

File: src/tokens/utilities.ts (hunk 1 above)
Comment (aaronccasanova) [reply src/tokens/utilities.ts:7]: image
In addition to the above screenshot, I verified in their GitHub API and FAQ wikis these are permalinks. I actually like having this reference as we can make sure the existing test cases still pass during future updates. That said, I'm also fine with removing the link as it's now checked into source control if we ever need it.

**Pull request merged**